### PR TITLE
GeometryPrecisionReducer: preserve collection-level of the input geometry 

### PR DESCRIPTION
**Pull request by pramsey**
References GH-662.

### CODE REVIEW

File: src/precision/GeometryPrecisionReducer.cpp [hunk 1]
```diff
@@ -53,6 +53,32 @@ GeometryPrecisionReducer::reduce(const Geometry& geom)
         reduced = PrecisionReducerTransformer::reduce(geom, targetPM, removeCollapsed);
     }
 
+    // Match the collection level of the output to the input
+    // if necessary
+    if (geom.isCollection()
+        && ! reduced->isCollection()
+        && (geom.getCoordinateDimension() == reduced->getCoordinateDimension())
+        && reduced->getGeometryTypeId() != GEOS_GEOMETRYCOLLECTION)
```
Comment (dbaston) [inline src/precision/GeometryPrecisionReducer.cpp:61]: This is implied by line 59, no?

File: src/precision/GeometryPrecisionReducer.cpp [hunk 2]
```diff
@@ -53,6 +53,32 @@ GeometryPrecisionReducer::reduce(const Geometry& geom)
...
+    {
+        GeometryTypeId typeId = reduced->getGeometryTypeId();
+        if (reduced->isEmpty()) return geom.getFactory()->createEmpty(geom.getGeometryTypeId());
+        std::vector<std::unique_ptr<Geometry>> subgeoms;
+        subgeoms.push_back(std::move(reduced));
```
Comment (dbaston) [inline src/precision/GeometryPrecisionReducer.cpp:66]: Thoughts on putting this logic into a method of `GeometryFactory` ?

File: src/precision/GeometryPrecisionReducer.cpp (hunk 1 above)
Comment (pramsey) [reply src/precision/GeometryPrecisionReducer.cpp:61]: Yes, true.

File: src/precision/GeometryPrecisionReducer.cpp (hunk 2 above)
Comment (pramsey) [reply src/precision/GeometryPrecisionReducer.cpp:66]: Makes sense... with a no-op pass through for already-collections?

**Comment by pramsey** [comment]
Suggestions from @dbaston incorporated.

**Pull request closed without merging**